common/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/listitem/DaxListItem.kt

```diff
@@ -0,0 +1,256 @@
+    internal abstract val trailingSwitch: SwitchView
+    internal abstract val betaPill: ImageView?
+    internal abstract val itemContainer: View
+
+    /** Sets the item click listener */
+    fun setClickListener(onClick: () -> Unit) {
+        itemContainer.setOnClickListener { onClick() }
+    }
+
+    /** Sets the primary text title */
+    fun setPrimaryText(title: String?) {
+        primaryText.text = title
+    }
+
+    /** Sets primary text color */
+    fun setPrimaryTextColor(@ColorRes colorRes: Int) {
+        primaryText.setTextColor(ContextCompat.getColorStateList(context, colorRes))
+    }
+
+    /** Sets primary text color state list */
+    fun setPrimaryTextColorStateList(stateList: ColorStateList?) {
+        primaryText.setTextColor(stateList)
+    }
+
+    /** Sets primary text truncation */
+    fun setPrimaryTextTruncation(truncated: Boolean) {
+        if (truncated) {
+            primaryText.maxLines = 1
+            primaryText.ellipsize = TruncateAt.END
+        } else {
+            primaryText.maxLines = Int.MAX_VALUE
+        }
+    }
+
+    /** Sets the secondary text title */
+    fun setSecondaryText(title: String?) {
+        secondaryText?.text = title
+    }
+
+    /** Sets secondary text color */
+    fun setSecondaryTextColor(@ColorRes colorRes: Int) {
+        secondaryText?.setTextColor(ContextCompat.getColorStateList(context, colorRes))
+    }
+
+    /** Sets secondary text color state list */
+    fun setSecondaryTextColorStateList(stateList: ColorStateList?) {
+        secondaryText?.setTextColor(stateList)
+    }
+
+    /** Sets the leading icon image drawable */
+    fun setLeadingIconDrawable(drawable: Drawable) {
+        leadingIcon.setImageDrawable(drawable)
+        setLeadingIconVisibility(true)
+    }
+
+    /** Sets the leading icon image resource */
+    fun setLeadingIconResource(@DrawableRes idRes: Int) {
+        leadingIcon.setImageResource(idRes)
+        setLeadingIconVisibility(true)
+    }
+
+    /** Sets the leading image content description */
+    fun setLeadingIconContentDescription(description: String) {
+        leadingIcon.contentDescription = description
+    }
+
+    /** Sets the leading icon image visibility */
+    fun setLeadingIconVisibility(visible: Boolean) {
+        if (visible) {
+            leadingIconContainer.show()
+        } else {
+            leadingIconContainer.gone()
+        }
+    }
+
+    /** Sets the leading icon background image type */
+    fun setLeadingIconBackgroundType(value: Int) {
+        if (value == 0) {
+            leadingIconContainer.setBackgroundResource(android.R.color.transparent)
+        }
+        if (value == 1) {
+            leadingIconContainer.setBackgroundResource(R.drawable.list_item_image_circular_background)
+        }
+        if (value == 2) {
+            leadingIconContainer.setBackgroundResource(R.drawable.list_item_image_round_background)
+        }
+        setLeadingIconVisibility(true)
+    }
```

Comment:
NIT: We can maybe create some constants instead using numbers directly. It would be nice to have if we need to set the background shape programatically from an external class.
Out of the scope of this PR though, so feel free to ignore me 😬

Reply:
Yeah I like that!